app/javascript/lib/validation/validate-file-content.js

```diff
@@ -356,12 +367,19 @@ export async function parseFile(file, fileType, fileOptions={}, sizeProps={}) {
     delimiter: null,
     isGzipped: null
   }
+
   const parseResult = { fileInfo, issues: [] }
 
   try {
     fileInfo.isGzipped = await validateGzipEncoding(file)
     // if the file is compressed or we can't figure out the compression, don't try to parse further
-    if (fileInfo.isGzipped || !PARSEABLE_TYPES.includes(fileType)) {
+    if (
+      !PARSEABLE_TYPES.includes(fileType) ||
+      fileInfo.isGzipped && (
+        file.size > sizeProps?.fileSizeTotal || // CSFV for gzip needs whole file
+        file.size >= MAX_GZIP_FILESIZE // See comment for MAX_GZIP_FILESIZE
+      )
+    ) {
```

Comment:
An intermediate variable will help make the if block less complex to read.  
```suggestion
    const isFileFragment = file.size > sizeProps?.fileSizeTotal // likely a partial download from a GCP bucket
    if (
      !PARSEABLE_TYPES.includes(fileType) ||
      fileInfo.isGzipped && (isFileFragment || file.size >= MAX_GZIP_FILESIZE)
      // current gunzip implementation needs a whole file; see comment for MAX_GZIP_FILESIZE
    ) {
```